ivy_tests/test_ivy/test_frontends/test_tensorflow/test_signal.py

```diff
@@ -1,36 +1,44 @@
+# global
 import ivy
-from hypothesis import given, strategies as st
+import numpy as np
+from hypothesis import given, assume, strategies as st
+from ivy.functional.backends.numpy.data_type import dtype
+
 # local
 import ivy_tests.test_ivy.helpers as helpers
 from ivy_tests.test_ivy.helpers import handle_cmd_line_args
 
-
 # hann_window
+
+
 @handle_cmd_line_args
 @given(
-    dtype_and_x=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("float")),
-    target_dtype=helpers.get_dtypes("float"),
-    periodic=st.booleans(),
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        num_arrays=2,
+        shared_dtype=True,
+    ),
     num_positional_args=helpers.num_positional_args(
-        fn_name="ivy.functional.frontends.tensorflow.signal.hann_window"
+        fn_name="ivy.functional.frontends.tensorflow.hann_window",
     ),
 )
-def test_tensorflow_hann_window(
-    dtype_and_x, target_dtype, as_variable, 
-    periodic, num_positional_args, native_array, fw
+def test_hann_window(
+    dtype_and_x,
+    as_variable,
+    num_positional_args,
+    native_array,
 ):
-    input_dtype, x = dtype_and_x
+    input_dtype, x , y = dtype_and_x
     helpers.test_frontend_function(
         input_dtypes=input_dtype,
         as_variable_flags=as_variable,
         with_out=False,
         num_positional_args=num_positional_args,
         native_array_flags=native_array,
-        fw=fw,
         frontend="tensorflow",
-        fn_tree="signal.hann_window",
-        window_length=x[0],
-        periodic=periodic,
-        dtype=target_dtype,
-    )
+        fn_tree="hann_window",
+        window_length=x,
+        periodic=True,
+        dtype=y
```

Comment:
the dtype here require a datatype not a number, and the periodic argument should be tested for both True and False.

Reply:
as presented here :https://www.tensorflow.org/api_docs/python/tf/signal/hann_window, the dtype argument takes in the data type to produce. Must be a floating point type, like ivy_tf.dtypes.float32.

Reply:
I changed the code was this resolved?

Reply:
yes, no problem now